packages/hub/src/hub.ts

```diff
@@ -105,7 +103,8 @@ export class Hub implements HubInterface {
    * @inheritDoc
    */
   public popScope(): boolean {
-    return this.getStack().pop() !== undefined;
+    if (this.getStack().length <= 1) return false;
+    return !!this.getStack().pop();
```

Comment:
This could be accessing `this._stack` directly, same in `pushScope`.

Why is there even a public `getStack()`?! Is there a use case to deal with stack layers from code external to this class?!